src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSourceBuilder.java

```diff
@@ -107,11 +111,22 @@ public final String repoOwner() {
         return repoOwner;
     }
 
+    /**
+     * Pass the credentials object to the {@link GitHubSCMSource}.
+     * @param credentials the {@link com.cloudbees.plugins.credentials.common.StandardCredentials}
+     * @return the current builder
+     */
+    @NonNull
+    SCMSourceBuilder<GitHubSCMSourceBuilder, GitHubSCMSource> withCredentials(StandardCredentials credentials) {
+        this.credentials = credentials;
+        return this;
+    }
+
     /** {@inheritDoc} */
     @NonNull
     @Override
     public GitHubSCMSource build() {
-        GitHubSCMSource result = new GitHubSCMSource(repoOwner, projectName());
+        GitHubSCMSource result = new GitHubSCMSource(repoOwner, projectName(), credentials);
```

Comment:
This looks to me that `credentials` and `credentials` could diverge. I think that `credentialsId()` should return `credentials.getId()` in case `credentials` is set. WDYT?

Reply:
Agreed. Would be more consistent to do this.

Reply:
@jeromepochat I applied your suggestion.